numba/core/inline_closurecall.py

```diff
@@ -599,7 +600,8 @@ def inline_closure_call(func_ir, glbls, block, i, callee, typingctx=None,
     var_dict = {}
     for var in callee_scope.localvars._con.values():
         if not (var.name in callee_code.co_freevars):
-            new_var = scope.redefine(mk_unique_var(var.name), loc=var.loc)
+            inlined_name = f'{callee_ir.func_id.unique_name}.{var.name}'
+            new_var = scope.redefine(inlined_name, loc=var.loc)
```

Comment:
Note: probably need to write a test for explicit closure inlining, it uses this path opposed to the function inliner.